### ISSUE TITLE
NullPointerException at OneSignalStateSynchronizer.java:266

### ISSUE THREAD

**Issue by DimaKoz** [OneSignal/OneSignal-Android-SDK#88]
I got this [crash.log](https://github.com/OneSignal/OneSignal-Android-SDK/files/417670/log.txt) many times.
build.gradle:
```Groovy 
compile 'com.onesignal:OneSignal:2.6.2@aar'
```
Devices info:
<img width="1435" alt="screen shot 2016-08-15 at 10 44 31" src="https://cloud.githubusercontent.com/assets/3092276/17658695/83c0ede0-62de-11e6-8e72-223bc52912c6.png">

**Comment by jkasten2** [comment]
@DimaKoz Thanks for the report. I'll review our code and report back.

Log contains:
```Java
Fatal Exception: java.lang.NullPointerException
       at com.onesignal.OneSignalStateSynchronizer$UserState.generateJsonDiff(OneSignalStateSynchronizer.java:266)
       at com.onesignal.OneSignalStateSynchronizer$UserState.access$700(OneSignalStateSynchronizer.java:208)
       at com.onesignal.OneSignalStateSynchronizer.syncUserState(OneSignalStateSynchronizer.java:485)
       at com.onesignal.SyncService$1.run(SyncService.java:62)
       at java.lang.Thread.run(Thread.java:856)
```

**Comment by jkasten2** [comment]
@DimaKoz Are you calling OneSignal init from an Activity or your Application class?
Have you deleted any users from our dashboard?
Or change the OneSignal app id in your app?

**Comment by jkasten2** [comment]
@DimaKoz I believe this issue is due to a race condition which I have guard against in commit 27347f17ab41719cc7eed53d1dfddf231777b391. It will be included in the 3.0.3 release.
